### PR TITLE
adding theme state

### PR DESCRIPTION
**Pull request by alexanderalkhazov**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -4,21 +4,27 @@ import { Routes, Route } from 'react-router-dom';
...
 import Forecasts from './pages/Forecasts';
-import { Box } from '@mui/material';
+import CssBaseline from '@mui/material/CssBaseline';
+import { Box, ThemeProvider } from '@mui/material';
+import { lightTheme, darkTheme } from './theme/themes';
+import { useSelector } from 'react-redux';
 
 const App = () => {
 
+  const currentTheme = useSelector((state: any) => state.themeSet.theme);
```
Comment (daniil322) [inline src/App.tsx:14]: add type to state

File: src/App.tsx [hunk 2]
```diff
@@ -4,21 +4,27 @@ import { Routes, Route } from 'react-router-dom';
...
+ 
   return (
-    <>
-      <Box>
+    <Box>
+      <ThemeProvider theme={currentTheme === 'light' ? lightTheme : darkTheme}>
```
Comment (daniil322) [inline src/App.tsx:18]: make light a constant

File: src/components/Forecast/Forecast.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ const Forecast = () => {
                         gutterBottom
                         variant='h5'
                     >
-                        Weather
+                        Weather Date 1/2/1998
```
Comment (daniil322) [inline src/components/Forecast/Forecast.tsx:28]: ?

File: src/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -72,19 +83,43 @@ const Navbar = () => {
                         horizontal: 'right'
                     }}
                 >
-                    <MenuItem onClick={handleCLose}>
-                        <Link to={'/'}>
-                            <Button color='inherit'>Current Location</Button>
+                    <MenuItem
```
Comment (daniil322) [inline src/components/Navbar/Navbar.tsx:86]: make an object and use map  you wrote the same code 3 times

File: src/components/ThemeSwitch/ThemeSwitch.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import { useDispatch } from "react-redux";
+import { toggleTheme } from "../../slices/themeSlice";
+
+const ThemeSwitch = () => {
+
+    const dispatch = useDispatch();
+
+    return (
+        <>
+            <Switch aria-label='switch' onClick={() => dispatch(toggleTheme())}/>
```
Comment (daniil322) [inline src/components/ThemeSwitch/ThemeSwitch.tsx:11]: do you even know what the aria means?

Comment (daniil322) [inline src/components/ThemeSwitch/ThemeSwitch.tsx:11]: and maybe make a hook where you return all of the actions from that store
useThemeActions

File: src/slices/themeSlice.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+const initialTheme = {
+    theme: 'light',
+};
+
+export const themeSlice = createSlice({
+    name: 'theme',
+    initialState: initialTheme,
+    reducers: {
+        toggleTheme: (state) => state.theme === 'light' ? {...state, theme: 'dark'} : {...state,theme:'light'}
```
Comment (daniil322) [inline src/slices/themeSlice.ts:11]: write the function outside so it will look cleaner

File: src/App.tsx (hunk 1 above)
Comment (alexanderalkhazov) [reply src/App.tsx:14]: ok i will check how

File: src/App.tsx (hunk 2 above)
Comment (alexanderalkhazov) [reply src/App.tsx:18]: ok done

File: src/components/Forecast/Forecast.tsx (hunk 3 above)
Comment (alexanderalkhazov) [reply src/components/Forecast/Forecast.tsx:28]: it's nothing dont worry about it

File: src/components/ThemeSwitch/ThemeSwitch.tsx (hunk 5 above)
Comment (alexanderalkhazov) [reply src/components/ThemeSwitch/ThemeSwitch.tsx:11]: yes, now i know

File: src/App.tsx (hunk 2 above)
Comment (alexanderalkhazov) [reply src/App.tsx:18]: but why tho ? seems pointless

File: src/components/Navbar/Navbar.tsx (hunk 4 above)
Comment (alexanderalkhazov) [reply src/components/Navbar/Navbar.tsx:86]: done !

File: src/components/ThemeSwitch/ThemeSwitch.tsx (hunk 5 above)
Comment (alexanderalkhazov) [reply src/components/ThemeSwitch/ThemeSwitch.tsx:11]: k done

File: src/slices/themeSlice.ts (hunk 6 above)
Comment (alexanderalkhazov) [reply src/slices/themeSlice.ts:11]: donr

File: src/App.tsx (hunk 2 above)
Comment (DaniilGL) [reply src/App.tsx:18]: To not write it everywhere

**Pull request merged**